apps/desktop-wallet/src/workers/addressDiscovery.ts

```diff
@@ -16,27 +16,28 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
+import { keyring } from '@alephium/keyring'
 import { exponentialBackoffFetchRetry } from '@alephium/shared'
-import { discoverActiveAddresses, Wallet, walletImport } from '@alephium/shared-crypto'
 import { ExplorerProvider } from '@alephium/web3'
 
 interface WorkerPayload {
   data: {
-    mnemonic: Wallet['mnemonic']
     clientUrl: string
     skipIndexes?: number[]
-    passphrase?: string
   }
 }
 
-self.onmessage = ({ data: { mnemonic, passphrase, clientUrl, skipIndexes } }: WorkerPayload) => {
-  const { masterKey } = walletImport(mnemonic, passphrase)
+self.onmessage = ({ data: { clientUrl, skipIndexes } }: WorkerPayload) => {
   const client = new ExplorerProvider(clientUrl, undefined, exponentialBackoffFetchRetry)
 
-  discover(masterKey, client, skipIndexes)
+  discover(client, skipIndexes)
 }
 
-const discover = async (masterKey: Wallet['masterKey'], client: ExplorerProvider, skipIndexes?: number[]) => {
-  const activeAddresses = await discoverActiveAddresses(masterKey, client, skipIndexes)
-  self.postMessage(activeAddresses)
+const discover = async (client: ExplorerProvider, skipIndexes?: number[]) => {
+  try {
+    const activeAddresses = await keyring.discoverAndCacheActiveAddresses(client, skipIndexes)
```

Comment:
Q: So this works inside of the worker? Wasn't there an issue with the keyring object not being the same in this context?

Reply:
Even though the web workers code has been updated, the use of it is commented out.

Reply:
Removed web workers completely in 0772f5c42 in PR #504